atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/async/client/creation/DefaultCqlClientFactory.java

```diff
@@ -47,10 +49,15 @@ public DefaultCqlClientFactory() {
     @Override
     public Optional<CqlClient> constructClient(
             TaggedMetricRegistry taggedMetricRegistry, CassandraKeyValueServiceConfig config, boolean initializeAsync) {
-        return CassandraServersConfigs.getCqlCapableConfigIfValid(config).map(cqlCapableConfig -> {
-            Set<InetSocketAddress> servers = cqlCapableConfig.cqlHosts();
-            Cluster cluster = new ClusterFactory(cqlClusterBuilderFactory).constructCluster(servers, config);
-            return CqlClientImpl.create(taggedMetricRegistry, cluster, cqlCapableConfig.tuning(), initializeAsync);
-        });
+        CassandraClusterConfig cassandraClusterConfig = CassandraClusterConfig.of(config);
+        CassandraServersConfig cassandraServersConfig = config.servers();
+        return CassandraServersConfigs.getCqlCapableConfigIfValid(cassandraServersConfig)
```

Comment:
This still takes in a config for now, since changing this would require changing a fair bit more code. I'll change this to take in only relevant information in another PR